### PR TITLE
Support testing against multiple minor versions of Scala

### PR DESCRIPTION
**Pull request by miloradvojnovic**
Summary:
 - Supported testing against multiple minor versions of Scala (2.12.x, 2.13.x)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,7 +36,12 @@ jobs:
         run: sbt compile Test/compile
 
       - name: Run tests
-        run: sbt test
+        run: |
+          sbt -scala-version 2.12.14 test
```
Comment (PawelLipski) [inline .github/workflows/ci.yml:40]: Hmmm actually... this fails for me locally, does it work on your machine? 🤔 

```
$ sbt -scala-version 2.13.6 test
[info] welcome to sbt 1.5.5 (Red Hat, Inc. Java 11.0.13)
[info] loading settings for project global-plugins from plugins.sbt ...
[info] loading global plugins from /home/plipski/.sbt/1.0/plugins
[info] loading settings for project akka-serialization-helper-build-build from plugins.sbt ...
[info] loading project definition from /home/plipski/akka-serialization-helper/project/project
[info] loading settings for project akka-serialization-helper-build from build.sbt,plugins.sbt ...
[info] loading project definition from /home/plipski/akka-serialization-helper/project
[info] loading settings for project akka-serialization-helper from build.sbt ...
[info] resolving key references (13734 settings) ...
[info] set current project to akka-serialization-helper (in build file:/home/plipski/akka-serialization-helper/)
[error] Expected symbol
[error] Not a valid command: -
[error] Expected end of input.
[error] Expected '--'
[error] Expected 'debug'
[error] Expected 'info'
[error] Expected 'warn'
[error] Expected 'error'
[error] Expected 'addPluginSbtFile'
[error] -scala-version
[error]  ^
```

Comment (miloradvojnovic) [reply .github/workflows/ci.yml:40]: No, it doesn't. I looked into CI logs and saw that we use `olafurpg/setup-scala@v10` which has additional args for sbt and one of those is `-scala-version`. Under the hood, it's `++ version` which is a regular sbt parameter for the version. So I used that for defining version.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -36,11 +36,21 @@ jobs:
...
+          sbt -scala-version 2.13.4 test
+          sbt -scala-version 2.13.5 test
+          sbt -scala-version 2.13.6 test
+          sbt -scala-version 2.13.7 test
+          sbt -scala-version 2.13.8 test
 
       - name: Check code style
         # Curiously, scalafmtCheckAll doesn't cover *.sbt files, hence scalafmtSbtCheck is needed as well
-        run: sbt scalafmtCheckAll scalafmtSbtCheck "scalafixAll --check"
+        run: sbt scalafmtCheckAll scalafmtSbtCheck # "scalafixAll --check"
```
Comment (PawelLipski) [inline .github/workflows/ci.yml:53]: Pls open a ticket to restore scalafix and reference it from here via a `TODO`

File: build.sbt [hunk 3]
```diff
@@ -21,8 +21,6 @@ sonatypeProfileName := "org.virtuslab"
 
 Global / onChangedBuildSource := ReloadOnSourceChanges
 
-ThisBuild / semanticdbEnabled := true
-ThisBuild / semanticdbVersion := scalafixSemanticdb.revision
```
Comment (PawelLipski) [inline build.sbt:25]: Pls comment out the scalafix plugin from plugins.sbt + add a TODO

File: .github/workflows/ci.yml (hunk 1 above)
Comment (PawelLipski) [reply .github/workflows/ci.yml:41]: Hmmm okay so could you change the syntax here for CI invocation so that it's portable to our local machines? or is `-scala-version` significantly distinct from `++ version`?

Comment (miloradvojnovic) [reply .github/workflows/ci.yml:41]: Sure. It's the same but `-scala-version <version>` is more readable so that's why I picked it over `++ <version>`.

Comment (PawelLipski) [reply .github/workflows/ci.yml:41]: Huh okay so at least add a comment how to reproduce the same test run locally (it's not gonna be obvious in a few months)... or is there a way to configure our local `sbt` so that `-scala-version` option is accepted as it is now in the CI?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -36,11 +36,21 @@ jobs:
...
+          sbt -scala-version 2.12.13 test
+          sbt -scala-version 2.12.14 test
+          sbt -scala-version 2.12.15 test
+          sbt -scala-version 2.13.2 test
+          sbt -scala-version 2.13.3 test
+          sbt -scala-version 2.13.4 test
+          sbt -scala-version 2.13.5 test
+          sbt -scala-version 2.13.6 test
+          sbt -scala-version 2.13.7 test
+          sbt -scala-version 2.13.8 test
```
Comment (MarconZet) [inline .github/workflows/ci.yml:49]: How long does it take now to complete CI?

I would advise splitting these tests into separate job

File: build.sbt (hunk 3 above)
Comment (MarconZet) [reply build.sbt:25]: Don't remove it, just change the sbt behaviour based on the selected version!!

File: dump-persistence-schema-compiler-plugin/src/test/resources/DataEnum.scala [hunk 5]
```diff
@@ -8,7 +8,7 @@ object DataEnum {
   case object Two extends Enum
   case object Three extends Enum
 
-  case class Data(enum: Enum)
+  case class Data(`enum`: Enum)
```
Comment (MarconZet) [inline dump-persistence-schema-compiler-plugin/src/test/resources/DataEnum.scala:11]: Just curious, why is this required?

File: .github/workflows/ci.yml (hunk 4 above)
Comment (MarconZet) [reply .github/workflows/ci.yml:49]: Or maybe add some sort of parallelism within the job (coping whole directory?)

Comment (miloradvojnovic) [reply .github/workflows/ci.yml:49]: It's increased from 6 to 24 minutes. We could have a separate job for every version.

File: dump-persistence-schema-compiler-plugin/src/test/resources/DataEnum.scala (hunk 5 above)
Comment (miloradvojnovic) [reply dump-persistence-schema-compiler-plugin/src/test/resources/DataEnum.scala:11]: It will become a keyword in Scala 3 (enum) and from Scala 2.13.7 there's a deprecation warning and because of that one test returns that as a result. 
```
"[warning: 1 deprecation (since 2.13.7); re-run with -deprecation for details [info]]" 
was not equal to "[]"
```

File: build.sbt (hunk 3 above)
Comment (PawelLipski) [reply build.sbt:25]: Yeah, even better!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (miloradvojnovic) [reply .github/workflows/ci.yml]: Changed to `++ <version>` because we can run it locally.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (miloradvojnovic) [reply .github/workflows/ci.yml:55]: #147

File: .github/workflows/ci.yml (hunk 4 above)
Comment (PawelLipski) [reply .github/workflows/ci.yml]: > separate job for every version

yeah, probably then the overhead of setting up a new docker container/compilation etc. will outweigh the benefits

Comment (miloradvojnovic) [reply .github/workflows/ci.yml]: I created separate jobs for testing. The first job is for testing version 2.12, the second and third is for 2.13. 
Before CI needed ~11 minutes, with sequential testing ~24min, and now ~10 minutes. You can check it and suggest if you have some other preferred way of grouping jobs/tasks.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -7,40 +7,115 @@ on:
   pull_request:
 
 jobs:
-  test:
+  initial-code-check:
```
Comment (PawelLipski) [inline .github/workflows/ci.yml:10]: This one is so lightweight that I think it can be merged back e.g. to `code-style-check`

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -7,40 +7,115 @@ on:
...
+      - name: Resolve dependencies
+        run: sbt update
+
+      - name: Compile
+        run: sbt compile Test/compile
+
+      - name: Run tests
+        run: |
+          sbt "++ 2.13.6 test"
+          sbt "++ 2.13.7 test"
```
Comment (PawelLipski) [inline .github/workflows/ci.yml:117]: Huh this one doesn't pass the CI, again due to scalafix... is it doable to make the version of `scalafix` sbt plugin (or the version of `semanticdb-scalac`?) conditional on `scala-version`?

Comment (miloradvojnovic) [reply .github/workflows/ci.yml:110]: I put a condition in `build.sbt` file to fix that.

File: build.sbt (hunk 3 above)
Comment (miloradvojnovic) [reply build.sbt:25]: I put a condition in `build.sbt` file to disable semanticdb if a scala version is `2.13.7` or `2.13.8`.

File: build.sbt [hunk 8]
```diff
@@ -21,7 +21,12 @@ sonatypeProfileName := "org.virtuslab"
...
 Global / onChangedBuildSource := ReloadOnSourceChanges
 
-ThisBuild / semanticdbEnabled := true
+ThisBuild / semanticdbEnabled := {
+  scalaVersion.value match {
+    case "2.13.7" | "2.13.8" => false
+    case _                   => true
+  }
+}
 ThisBuild / semanticdbVersion := scalafixSemanticdb.revision
```
Comment (PawelLipski) [inline build.sbt:30]:
```suggestion
ThisBuild / semanticdbEnabled := true
ThisBuild / semanticdbVersion := {
  scalaVersion.value match {
    case "2.13.7" | "2.13.8" => "4.4.32"
    case _                   => "4.4.28"
  }
}
```

Comment (PawelLipski) [reply build.sbt:30]: Seems to do the job for me locally

Comment (miloradvojnovic) [reply build.sbt]: Yes. It works now on all versions and CI jobs pass also.

**Review by PawelLipski — APPROVED** [review]

**Comment by PawelLipski** [comment]
@miloradvojnovic merge?
